File: GroupableGridView/Comparers/DataRowComparer.vb
Imports System.ComponentModel

''' <summary>
''' 
''' </summary>
Public Class DataRowComparer
	Implements IComparer(Of DataRow)

#Region " Objects and variables "

	Private m_Direction As ListSortDirection
	Private m_ColumnIndex As Integer

#End Region

#Region " Public methods and functions "

	Public Function Compare(x As DataRow, y As DataRow) As Integer Implements IComparer(Of DataRow).Compare

		Return String.Compare(x(m_ColumnIndex).ToString, y(m_ColumnIndex).ToString) * If(m_Direction = ListSortDirection.Ascending, 1, -1)

	End Function

#End Region

#Region " Construction "

	Public Sub New(columnIndex As Integer, sortDirection As ListSortDirection)

		m_Direction = sortDirection
		m_ColumnIndex = columnIndex

	End Sub

#End Region

End Class

File: GroupableGrid/OutlookGrid/DataRowComparer.cs
﻿using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/// <summary>
/// reusable custom DataRow comparer implementation, can be used to sort DataTables
/// </summary>
public class DataRowComparer : IComparer
{
	ListSortDirection direction;
	int columnIndex;

	public DataRowComparer(int columnIndex, ListSortDirection direction)
	{
		this.columnIndex = columnIndex;
		this.direction = direction;
	}

	#region IComparer Members

	public int Compare(object x, object y)
	{

		DataRow obj1 = (DataRow)x;
		DataRow obj2 = (DataRow)y;
		return string.Compare(obj1[columnIndex].ToString(), obj2[columnIndex].ToString()) * (direction == ListSortDirection.Ascending ? 1 : -1);
	}

	#endregion
}